typed-racket-lib/typed-racket/base-env/prims-contract.rkt

```diff
@@ -250,9 +251,17 @@
 ;;  make-predicate
 ;;  cast
 
-;; Helper to construct syntax for contract definitions
+;; Helpers to construct syntax for contract definitions
+;; make-contract-def-rhs : Type-Stx Boolean Boolean -> Syntax
 (define (make-contract-def-rhs type flat? maker?)
-  (contract-def-property #'#f `#s(contract-def ,type ,flat? ,maker? untyped)))
+  (define contract-def `#s(contract-def ,type ,flat? ,maker? untyped))
+  (contract-def-property #'#f (λ () contract-def)))
```

Comment:
No, because if type-checking somehow stops in a way that it never sees the `casted-expr` property, then it will seem like a too-early force, and in that case it makes sense to generate the wrapping `Any` contract as a fallback.